src/EventListener/EndEmptySessionSubscriber.php

```diff
@@ -22,13 +23,15 @@ public function onKernelResponse(FilterResponseEvent $event)
         $request = $event->getRequest();
         $session = $request->getSession();
 
-        if (!$session || (!$session->isStarted() && !$request->hasPreviousSession())) {
+        if (!$session || (!$session->isStarted() && !$request->hasPreviousSession()) || !empty($session->all())) {
             return;
         }
 
-        if (empty($session->all())) {
-            $session->invalidate();
-            $event->getResponse()->headers->clearCookie($session->getName());
+        if ($session instanceof Session && !empty($session->getFlashBag()->peekAll())) {
```

Comment:
There can be other bags which might not be empty (not that we use any), can't see a way to see what bags a session contains...